static/elements/chromedash-featurelist.html

```diff
@@ -130,38 +178,72 @@
           }
           this.filtered = this.features;
         } else {
+
           val = val.trim();
           if (history && history.replaceState) {
             history.replaceState({id: null}, document.title, '/features#' + val);
           }
 
-          // owner: user@chromium.org
-          var ownerMatch = val.match(/^owner:\s*(.*)/);
-          // category: Multimedia
-          var categoryMatch = val.match(/^category:\s*(.*)/);
-          // Returns operator and version query e.g. ["<=25", "<=", "25"].
-          var operatorMatch = /^([<>=]=?)\s*([0-9]+)/.exec(val);
-
-          if (ownerMatch) {
-            this.filtered = this._filterOnOwner(ownerMatch[1]);
-          } else if (categoryMatch) {
-            this.filtered = this._filterOnCategory(categoryMatch[1]);
-          } else if (operatorMatch) {
-            this.filtered = this._filterOnOperation(operatorMatch[1], operatorMatch[2]);
+          var val = ' foo "bar baz" quz " quuz zzz"';
+          var regExp = /"([^"]*)"/g;
+          var start = 0;
+          var parts = [];
+          while ((match = regExp.exec(val)) !== null) {
+            if (start - (regExp.lastIndex - match[0].length) > 0) {
+              parts.push(val.substring(start, regExp.lastIndex -
+                                       match[0].length));
+            }
+            parts.push(match[1]);
+          }
+          if (start - (regExp.lastIndex - match[0].length) > 0) {
+            parts.push(val.substring(start, regExp.lastIndex -
+                                     match[0].length));
+          }
+
+          // Match words separated by whitespace and/or ":".
+          regExp = /([^:<>= \f\n\r\t\v\u00a0\u1680\u180e\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]+)([:<>= \f\n\r\t\v\u00a0\u1680\u180e\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]+)?/g;
+          var match;
+          var reMatches = [];
+          while (match = regExp.exec(val) !== null) {
+            reMatches.push(match);
+          }
+
+          // Accumulate keyword and property queries.
+          var propertyQueries = [];
+          var operatorQueries = [];
+          var keywordQueries = [];
+          var i;
+          for (i = 0; i < reMatches.length; i++) {
+            match = reMatches[i];
+            if (match[2] && match[2].match(':') !== null &&
+                i < reMatches.length - 1) {
+              propertyQueries.push([match[1], reMatches[i + 1][1]]);
+              i++;
+            } else if (match[2] && match[2].match(/[<>=]/) !== null &&
+                       i < reMatches.length - 1) {
+              operatorQueries.push([match[1], match[2], reMatches[i + 1][1]]);
+              i++;
+            } else {
+              keywordQueries.push(match[1]);
+            }
+          }
+
+          var filters = propertyQueries.map(
```

Comment:
This is hard to grok. Can you add comments and maybe have separate lines with good var descriptions?